File: svws-core/src/main/java/de/svws_nrw/core/data/kataloge/OrtKatalogEintrag.java
package de.svws_nrw.core.data.kataloge;

import jakarta.xml.bind.annotation.XmlRootElement;
import de.svws_nrw.transpiler.TranspilerDTO;
import io.swagger.v3.oas.annotations.media.Schema;

/**
 * Diese Klasse wird bei der Kommunikation über die Open-API-Schnittstelle verwendet.
 * Sie beschreibt, wie die Daten für die Katalog der Orte übergeben werden.
 */
@XmlRootElement
@Schema(description = "ein Eintrag in dem Katalog der Orte.")
@TranspilerDTO
public class OrtKatalogEintrag {

	/** Die ID des Katalog-Eintrags. */
	@Schema(description = "die ID des Katalog-Eintrags", example = "4711")
	public long id;

	/** Die Postleitzahl. */
	@Schema(description = "die Postleitzahl", example = "53840")
	public String plz;

	/** Der Name des Ortes. */
	@Schema(description = "der Name des Ortes", example = "Troisdorf")
	public String ortsname;

	/** Der Name des Kreises. */
	@Schema(description = "der Name des Kreises", example = "Rhein-Sieg-Kreis")
	public String kreis;

	/** Das Kürzel des Bundeslandes.  */
	@Schema(description = "das Kürzel des Bundeslandes", example = "NW")
	public String kuerzelBundesland;

	/** Gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an. */
	@Schema(description = "gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an", example = "1")
	public int sortierung;

	/** Gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht. */
	@Schema(description = "gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht", example = "true")
	public boolean istSichtbar;

	/** Gibt an, ob der Eintrag in der Anwendung änderbar sein soll oder nicht. */
	@Schema(description = "gibt an, ob der Eintrag in der Anwendung änderbar sein soll oder nicht", example = "true")
	public boolean istAenderbar;

	/**
	 * Leerer Standardkonstruktor.
	 */
	public OrtKatalogEintrag() {
		// leer
	}

}

File: svws-webclient/core/src/core/data/kataloge/OrtKatalogEintrag.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { Class } from '../../../java/lang/Class';

export class OrtKatalogEintrag extends JavaObject {

	/**
	 * Die ID des Katalog-Eintrags.
	 */
	public id : number = 0;

	/**
	 * Die Postleitzahl.
	 */
	public plz : string | null = null;

	/**
	 * Der Name des Ortes.
	 */
	public ortsname : string | null = null;

	/**
	 * Der Name des Kreises.
	 */
	public kreis : string | null = null;

	/**
	 * Das Kürzel des Bundeslandes.
	 */
	public kuerzelBundesland : string | null = null;

	/**
	 * Gibt die Position in der Sortierreihenfolge für die Katalog-Einträge an.
	 */
	public sortierung : number = 0;

	/**
	 * Gibt an, ob der Eintrag in der Anwendung sichtbar sein soll oder nicht.
	 */
	public istSichtbar : boolean = false;

	/**
	 * Gibt an, ob der Eintrag in der Anwendung änderbar sein soll oder nicht.
	 */
	public istAenderbar : boolean = false;


	public constructor() {
		super();
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.data.kataloge.OrtKatalogEintrag';
	}

	isTranspiledInstanceOf(name : string): boolean {
		return ['de.svws_nrw.core.data.kataloge.OrtKatalogEintrag'].includes(name);
	}

	public static class = new Class<OrtKatalogEintrag>('de.svws_nrw.core.data.kataloge.OrtKatalogEintrag');

	public static transpilerFromJSON(json : string): OrtKatalogEintrag {
		const obj = JSON.parse(json) as Partial<OrtKatalogEintrag>;
		const result = new OrtKatalogEintrag();
		if (obj.id === undefined)
			throw new Error('invalid json format, missing attribute id');
		result.id = obj.id;
		result.plz = (obj.plz === undefined) ? null : obj.plz === null ? null : obj.plz;
		result.ortsname = (obj.ortsname === undefined) ? null : obj.ortsname === null ? null : obj.ortsname;
		result.kreis = (obj.kreis === undefined) ? null : obj.kreis === null ? null : obj.kreis;
		result.kuerzelBundesland = (obj.kuerzelBundesland === undefined) ? null : obj.kuerzelBundesland === null ? null : obj.kuerzelBundesland;
		if (obj.sortierung === undefined)
			throw new Error('invalid json format, missing attribute sortierung');
		result.sortierung = obj.sortierung;
		if (obj.istSichtbar === undefined)
			throw new Error('invalid json format, missing attribute istSichtbar');
		result.istSichtbar = obj.istSichtbar;
		if (obj.istAenderbar === undefined)
			throw new Error('invalid json format, missing attribute istAenderbar');
		result.istAenderbar = obj.istAenderbar;
		return result;
	}

	public static transpilerToJSON(obj : OrtKatalogEintrag) : string {
		let result = '{';
		result += '"id" : ' + obj.id.toString() + ',';
		result += '"plz" : ' + ((!obj.plz) ? 'null' : JSON.stringify(obj.plz)) + ',';
		result += '"ortsname" : ' + ((!obj.ortsname) ? 'null' : JSON.stringify(obj.ortsname)) + ',';
		result += '"kreis" : ' + ((!obj.kreis) ? 'null' : JSON.stringify(obj.kreis)) + ',';
		result += '"kuerzelBundesland" : ' + ((!obj.kuerzelBundesland) ? 'null' : JSON.stringify(obj.kuerzelBundesland)) + ',';
		result += '"sortierung" : ' + obj.sortierung.toString() + ',';
		result += '"istSichtbar" : ' + obj.istSichtbar.toString() + ',';
		result += '"istAenderbar" : ' + obj.istAenderbar.toString() + ',';
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

	public static transpilerToJSONPatch(obj : Partial<OrtKatalogEintrag>) : string {
		let result = '{';
		if (obj.id !== undefined) {
			result += '"id" : ' + obj.id.toString() + ',';
		}
		if (obj.plz !== undefined) {
			result += '"plz" : ' + ((!obj.plz) ? 'null' : JSON.stringify(obj.plz)) + ',';
		}
		if (obj.ortsname !== undefined) {
			result += '"ortsname" : ' + ((!obj.ortsname) ? 'null' : JSON.stringify(obj.ortsname)) + ',';
		}
		if (obj.kreis !== undefined) {
			result += '"kreis" : ' + ((!obj.kreis) ? 'null' : JSON.stringify(obj.kreis)) + ',';
		}
		if (obj.kuerzelBundesland !== undefined) {
			result += '"kuerzelBundesland" : ' + ((!obj.kuerzelBundesland) ? 'null' : JSON.stringify(obj.kuerzelBundesland)) + ',';
		}
		if (obj.sortierung !== undefined) {
			result += '"sortierung" : ' + obj.sortierung.toString() + ',';
		}
		if (obj.istSichtbar !== undefined) {
			result += '"istSichtbar" : ' + obj.istSichtbar.toString() + ',';
		}
		if (obj.istAenderbar !== undefined) {
			result += '"istAenderbar" : ' + obj.istAenderbar.toString() + ',';
		}
		result = result.slice(0, -1);
		result += '}';
		return result;
	}

}

export function cast_de_svws_nrw_core_data_kataloge_OrtKatalogEintrag(obj : unknown) : OrtKatalogEintrag {
	return obj as OrtKatalogEintrag;
}
